### PR TITLE
Mushfichowdhury sc 61149 implement event emitting in magic sdk

### PR DESCRIPTION
**Pull request by mushfichowdhury-magic**
### 📦 Pull Request

Bidirectional RPC Event Support

### CODE REVIEW

File: packages/@magic-sdk/provider/test/spec/modules/base-module/request.spec.ts [hunk 1]
```diff
@@ -203,3 +203,104 @@ test('Ignores events with malformed response', (done) => {
...
+        },
+      ],
+    },
+  ];
+  const response = new JsonRpcResponse(requestPayload).applyResult(expectedResult);
+  const mock = jest.fn().mockImplementation(() => Promise.resolve(response));
+  const { baseModule } = createBaseModule(mock);
+  const intermediaryEventFunc = baseModule.createIntermediaryEvent('event', requestPayload.id);
+  intermediaryEventFunc(null);
+  console.log(mock.mock.calls[0]);
```
Comment (dgerrellsMagic) [inline packages/@magic-sdk/provider/test/spec/modules/base-module/request.spec.ts:304]: We can remove the console here. :)

**Review by dgerrellsMagic — APPROVED** [review]

File: packages/@magic-sdk/provider/test/spec/modules/base-module/request.spec.ts [hunk 2]
```diff
@@ -203,3 +203,103 @@ test('Ignores events with malformed response', (done) => {
...
 });
+
+test('Resolves with an intermediary event', async () => {
+  const expectedResult = 'monkey brains';
+  const expectedPayload = [
+    'MAGIC_HANDLE_REQUEST',
+    {
+      id: (requestPayload.id as number) + 1,
+      jsonrpc: requestPayload.jsonrpc,
+      method: 'magic_intermediary_event',
```
Comment (Ethella) [inline packages/@magic-sdk/provider/test/spec/modules/base-module/request.spec.ts:214]: QQ, how would the SDK know which root method the intermediary event is associated with?

File: packages/@magic-sdk/provider/src/modules/auth.ts [hunk 3]
```diff
@@ -53,9 +49,9 @@ export class AuthModule extends BaseModule {
     const { email } = configuration;
     const requestPayload = createJsonRpcRequestPayload(
       this.sdk.testMode ? MagicPayloadMethod.LoginWithEmailOTPTestMode : MagicPayloadMethod.LoginWithEmailOTP,
-      [{ email, showUI: true }],
```
Comment (Ethella) [inline packages/@magic-sdk/provider/src/modules/auth.ts:56]: With this change, EmailOTP can hide the pop window now. Am I correct?

File: packages/@magic-sdk/provider/test/spec/modules/base-module/request.spec.ts (hunk 2 above)
Comment (dgerrellsMagic) [reply packages/@magic-sdk/provider/test/spec/modules/base-module/request.spec.ts:214]: The arguments for the call include the id of the root method in it.

File: packages/@magic-sdk/provider/src/modules/auth.ts (hunk 3 above)
Comment (dgerrellsMagic) [reply packages/@magic-sdk/provider/src/modules/auth.ts:56]: That is correct. Good catch @Ethella

**Comment by Ethella** [comment]
Thx for answering. LGTM

**Review by Ethella — APPROVED** [review]

**Comment by magiclabsFE** [comment]
:rocket: PR was released in [`@magic-ext/algorand@6.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/algorand%406.0.0), [`@magic-ext/avalanche@6.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/avalanche%406.0.0), [`@magic-ext/bitcoin@6.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/bitcoin%406.0.0), [`@magic-ext/conflux@4.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/conflux%404.0.0), [`@magic-ext/connect@4.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/connect%404.0.0), [`@magic-ext/cosmos@6.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/cosmos%406.0.0), [`@magic-ext/ed25519@2.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/ed25519%402.0.0), [`@magic-ext/flow@6.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/flow%406.0.0), [`@magic-ext/harmony@6.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/harmony%406.0.0), [`@magic-ext/icon@6.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/icon%406.0.0), [`@magic-ext/near@6.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/near%406.0.0), [`@magic-ext/oauth@5.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/oauth%405.0.0), [`@magic-ext/polkadot@6.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/polkadot%406.0.0), [`@magic-ext/react-native-oauth@5.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/react-native-oauth%405.0.0), [`@magic-ext/solana@7.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/solana%407.0.0), [`@magic-ext/taquito@3.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/taquito%403.0.0), [`@magic-ext/terra@3.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/terra%403.0.0), [`@magic-ext/tezos@6.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/tezos%406.0.0), [`@magic-ext/webauthn@5.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/webauthn%405.0.0), [`@magic-ext/zilliqa@6.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/zilliqa%406.0.0), [`@magic-sdk/commons@7.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/commons%407.0.0), [`@magic-sdk/pnp@5.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/pnp%405.0.0), [`@magic-sdk/provider@11.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/provider%4011.0.0), [`@magic-sdk/react-native@12.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/react-native%4012.0.0), [`@magic-sdk/types@10.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/types%4010.0.0), [`magic-sdk@11.0.0`](https://github.com/magiclabs/magic-js/releases/tag/magic-sdk%4011.0.0) :rocket:

**Pull request merged**